ballerina/tests/client_test.bal

```diff
@@ -0,0 +1,445 @@
+// Copyright (c) 2022, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+//
+// WSO2 Inc. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+import ballerina/test;
+// import ballerina/io;
+
+@test:Config {
+    groups: ["client"]
+}
+function testExecuteWithTypeWithJson() returns error? {
+    // GraphQL API URL
+    string url = "https://countries.trevorblades.com/";
+    // Query
+    string query = string `query countryByCode($code:ID!) {country(code:$code) {name}}`;
+    // Variables
+    string code = "LK";
+    map<anydata> variables = {"code": code};
+
+    Client graphqlClient = check new (url);
+    json actualPayload = check graphqlClient->executeWithType(query, variables);
+    string actual = actualPayload.toJsonString();
+
+    json expectedPayload = {
+        "data": {
+            "country": {
+                "name": "Sri Lanka"
+            }
+        }
+    };
+    string expected = expectedPayload.toJsonString();
+
+    test:assertEquals(actual, expected);
```

Comment:
There's already a util method (`assertJsonValuesWithOrder`) to assert two JSON values. We do not need the `toJsonString` method here.

Reply:
assertJsonValuesWithOrder util method is used. The changes are amended. Thank you.